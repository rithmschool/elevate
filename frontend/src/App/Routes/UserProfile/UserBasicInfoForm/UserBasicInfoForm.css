/**/
.borderrounded {
  margin: 10% auto;
  background-color: #f4f6f8;
}
