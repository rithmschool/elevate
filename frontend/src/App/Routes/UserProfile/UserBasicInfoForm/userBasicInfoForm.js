import React, { useState } from "react";
import { Button, Form } from "react-bootstrap";
import moment from "moment";
import "../FormStyles.css";
import "./userBasicInfoForm.scss";

/** Update user basic info */
function UserBasicInfoForm(props) {
  const [first_name, setFirstName] = useState(props.first_name);
  const [last_name, setLastName] = useState(props.last_name);
  const [email, setEmail] = useState(props.email);
  const [current_company, setCurrentCompany] = useState(props.current_company);
  const [hire_date, setHireDate] = useState(props.hire_date);
  const [isEdit, setIsEdit] = useState(false);

  return (
    <div
      className={`user-basic-info-form container border rounded shadow pt-5`}
    >
      <Form onSubmit={handleSubmit}>
        <div className="form-styles_flex-space-between">
          <h3>Basic info</h3>

          {!isEdit && (
            <i
              data-testid="turnEditOn"
              className="m-3 fas fa-edit fa-1x"
              onClick={() => setIsEdit(true)}
            ></i>
          )}
        </div>

        <Form.Group>
          <span>First name</span>

          <Form.Control
            onChange={e => setFirstName(e.target.value)}
            id="EditUser-first_name"
            data-testid="firstName"
            name="first_name"
            type="text"
            disabled={!isEdit}
            value={first_name}
          />
        </Form.Group>

        <Form.Group>
          <span>Last name</span>

          <Form.Control
            onChange={e => setLastName(e.target.value)}
            id="EditUser-last_name"
            data-testid="lastName"
            name="last_name"
            type="text"
            disabled={!isEdit}
            value={last_name}
          />
        </Form.Group>

        <Form.Group>
          <span>Email</span>

          <Form.Control
            onChange={e => setEmail(e.target.value)}
            id="EditUser-email"
            data-testid="email"
            name="email"
            type="text"
            disabled={!isEdit}
            value={email}
          />
        </Form.Group>

        <Form.Group>
          <span>Current company</span>

          <Form.Control
            onChange={e => setCurrentCompany(e.target.value)}
            id="EditUser-current_company"
            data-testid="currentCompany"
            name="current_company"
            type="text"
            disabled={!isEdit}
            value={current_company}
          />
        </Form.Group>

        <Form.Group>
          <span>Hire date</span>

          <Form.Control
            onChange={e => setHireDate(e.target.value)}
            id="EditUser-hire_date"
            data-testid="hireDate"
            name="hire_date"
            type="date"
            disabled={!isEdit}
            value={formatHireDate() || ""}
          />
        </Form.Group>

        <div className="row justify-content-center">
          {isEdit && (
            <div>
              <Button className="login-submit mr-3 ml-3" type="submit">
                Submit
              </Button>

              <h6
                className="mr-3 ml-3 form-styles_cancel"
                data-testid="cancel"
                onClick={() => setIsEdit(false)}
              >
                Cancel
              </h6>
            </div>
          )}
        </div>
      </Form>
    </div>
  );

  function formatHireDate() {
    if (hire_date === null) {
      return null;
    } else {
      return moment(hire_date).format("YYYY-MM-DD");
    }
  }

  function handleSubmit(evt) {
    evt.preventDefault();

    const updatedUser = {
      first_name,
      last_name,
      email,
      current_company,
      hire_date: formatHireDate()
    };

    props.handleUserUpdate(updatedUser, props.userId);
    setIsEdit(false);
  }
}

export default UserBasicInfoForm;
