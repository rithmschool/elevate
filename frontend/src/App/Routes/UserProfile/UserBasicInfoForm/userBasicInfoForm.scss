.user-basic-info-form {
  .border.rounded {
    margin: 10% auto;
    background-color: #f4f6f8;
  }
}
