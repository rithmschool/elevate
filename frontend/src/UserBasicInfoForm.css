.user-basic-info-form_cancel {
  padding: 0.8em 0;
  color: #555;
  cursor: pointer;
}

.user-basic-info-form_flex-space-between {
  display: flex;
  justify-content: space-between;
}
