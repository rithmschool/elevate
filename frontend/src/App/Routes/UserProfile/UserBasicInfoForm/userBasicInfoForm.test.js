import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import { render, fireEvent } from "@testing-library/react";
import { Form } from "react-bootstrap";
import UserBasicInfoForm from "./userBasicInfoForm";

describe("UserBasicInfoForm", function() {
  let wrapper;
  let currentUser = {
    current_company: "Rithm",
    email: "test@gmail.com",
    first_name: "user",
    hire_date: "2019-08-13",
    last_name: "test"
  };

  beforeEach(function() {
    wrapper = mount(<UserBasicInfoForm {...currentUser} />);
  });

  it("renders without crashing", function() {
    render(<UserBasicInfoForm {...currentUser} />);
  });

  it("matches snapshot", function() {
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it("renders all inputs", function() {
    expect(wrapper.find(Form.Control)).toHaveLength(5);
  });

  it("renders correct initial state in form", function() {
    const { getByTestId } = render(<UserBasicInfoForm {...currentUser} />);
    const firstNameInput = getByTestId("firstName");
    const lastNameInput = getByTestId("lastName");
    const emailInput = getByTestId("email");
    const companyInput = getByTestId("currentCompany");
    const hireDateInput = getByTestId("hireDate");
    expect(firstNameInput.value).toBe("user");
    expect(lastNameInput.value).toBe("test");
    expect(emailInput.value).toBe("test@gmail.com");
    expect(companyInput.value).toBe("Rithm");
    expect(hireDateInput.value).toBe("2019-08-13");
  });

  it("changes values via the fireEvent.change method", function() {
    const { getByTestId } = render(<UserBasicInfoForm {...currentUser} />);
    const input = getByTestId("firstName");
    fireEvent.change(input, { target: { value: "a" } });
    expect(input.value).toBe("a");
    fireEvent.change(input, { target: { value: "aladdin" } });
    expect(input.value).toBe("aladdin");
  });

  it("enables the form inputs when edit toggled on", function() {
    const { getByTestId } = render(<UserBasicInfoForm {...currentUser} />);
    const input = getByTestId("firstName");
    const toggle = getByTestId("turnEditOn");
    fireEvent.click(toggle);
    expect(input.disabled).toBeFalsy();
  });

  it("disables the form inputs on submission", function() {
    const handleUserUpdate = jest.fn();
    const { getByTestId } = render(
      <UserBasicInfoForm {...currentUser} handleUserUpdate={handleUserUpdate} />
    );
    const input = getByTestId("firstName");
    fireEvent.submit(input);
    expect(input.disabled).toBeTruthy();
  });

  it("disables the form inputs upon clicking cancel", function() {
    const handleUserUpdate = jest.fn();
    const { getByTestId } = render(
      <UserBasicInfoForm {...currentUser} handleUserUpdate={handleUserUpdate} />
    );
    const toggle = getByTestId("turnEditOn");
    const input = getByTestId("firstName");
    fireEvent.click(toggle);
    expect(input.disabled).toBeFalsy();
    const cancel = getByTestId("cancel");
    fireEvent.click(cancel);
    expect(input.disabled).toBeTruthy();
  });

  it("calls handleUserUpdate on submission", function() {
    const handleUserUpdate = jest.fn();
    const { getByTestId } = render(
      <UserBasicInfoForm {...currentUser} handleUserUpdate={handleUserUpdate} />
    );
    const input = getByTestId("firstName");
    fireEvent.submit(input);
    expect(handleUserUpdate).toBeCalledTimes(1);
  });
});
