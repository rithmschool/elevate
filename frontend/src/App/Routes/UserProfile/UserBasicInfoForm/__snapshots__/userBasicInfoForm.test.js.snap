// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBasicInfoForm matches snapshot 1`] = `
<UserBasicInfoForm
  current_company="Rithm"
  email="test@gmail.com"
  first_name="user"
  hire_date="2019-08-13"
  last_name="test"
>
  <div
    className="container borderrounded shadow"
  >
    <div
      className="form-inside-container mt-5"
    >
      <Form
        inline={false}
        onSubmit={[Function]}
      >
        <form
          className=""
          onSubmit={[Function]}
        >
          <div
            className="form-styles_flex-space-between"
          >
            <h3>
              Basic info
            </h3>
            <i
              className="m-3 fas fa-edit fa-1x"
              data-testid="turnEditOn"
              onClick={[Function]}
            />
          </div>
          <FormGroup>
            <div
              className="form-group"
            >
              <span>
                First name
              </span>
              <FormControl
                data-testid="firstName"
                disabled={true}
                id="EditUser-first_name"
                name="first_name"
                onChange={[Function]}
                type="text"
                value="user"
              >
                <input
                  className="form-control"
                  data-testid="firstName"
                  disabled={true}
                  id="EditUser-first_name"
                  name="first_name"
                  onChange={[Function]}
                  type="text"
                  value="user"
                />
              </FormControl>
            </div>
          </FormGroup>
          <FormGroup>
            <div
              className="form-group"
            >
              <span>
                Last name
              </span>
              <FormControl
                data-testid="lastName"
                disabled={true}
                id="EditUser-last_name"
                name="last_name"
                onChange={[Function]}
                type="text"
                value="test"
              >
                <input
                  className="form-control"
                  data-testid="lastName"
                  disabled={true}
                  id="EditUser-last_name"
                  name="last_name"
                  onChange={[Function]}
                  type="text"
                  value="test"
                />
              </FormControl>
            </div>
          </FormGroup>
          <FormGroup>
            <div
              className="form-group"
            >
              <span>
                Email
              </span>
              <FormControl
                data-testid="email"
                disabled={true}
                id="EditUser-email"
                name="email"
                onChange={[Function]}
                type="text"
                value="test@gmail.com"
              >
                <input
                  className="form-control"
                  data-testid="email"
                  disabled={true}
                  id="EditUser-email"
                  name="email"
                  onChange={[Function]}
                  type="text"
                  value="test@gmail.com"
                />
              </FormControl>
            </div>
          </FormGroup>
          <FormGroup>
            <div
              className="form-group"
            >
              <span>
                Current company
              </span>
              <FormControl
                data-testid="currentCompany"
                disabled={true}
                id="EditUser-current_company"
                name="current_company"
                onChange={[Function]}
                type="text"
                value="Rithm"
              >
                <input
                  className="form-control"
                  data-testid="currentCompany"
                  disabled={true}
                  id="EditUser-current_company"
                  name="current_company"
                  onChange={[Function]}
                  type="text"
                  value="Rithm"
                />
              </FormControl>
            </div>
          </FormGroup>
          <FormGroup>
            <div
              className="form-group"
            >
              <span>
                Hire date
              </span>
              <FormControl
                data-testid="hireDate"
                disabled={true}
                id="EditUser-hire_date"
                name="hire_date"
                onChange={[Function]}
                type="date"
                value="2019-08-13"
              >
                <input
                  className="form-control"
                  data-testid="hireDate"
                  disabled={true}
                  id="EditUser-hire_date"
                  name="hire_date"
                  onChange={[Function]}
                  type="date"
                  value="2019-08-13"
                />
              </FormControl>
            </div>
          </FormGroup>
          <div
            className="row justify-content-center"
          />
        </form>
      </Form>
    </div>
  </div>
</UserBasicInfoForm>
`;
